packages/twenty-server/src/core/open-api/utils/components.utils.ts

```diff
@@ -0,0 +1,81 @@
+import { ObjectMetadataEntity } from 'src/metadata/object-metadata/object-metadata.entity';
+import { FieldMetadataType } from 'src/metadata/field-metadata/field-metadata.entity';
+
+export const computeSchemaComponents = (
+  item: ObjectMetadataEntity,
+): {
+  type: string;
+  required?: string[];
+  properties: { [key: string]: { type: string; items: object } };
+} => {
+  const result = {
+    type: 'object',
+    properties: item.fields.reduce((node, field) => {
+      const itemProperties = {
+        type: 'string',
+      };
+
+      if (
+        field.type === FieldMetadataType.NUMBER ||
+        field.type === FieldMetadataType.NUMERIC ||
+        field.type === FieldMetadataType.PROBABILITY ||
+        field.type === FieldMetadataType.RATING
+      ) {
+        itemProperties.type = 'number';
+      }
+
+      if (field.type === FieldMetadataType.BOOLEAN) {
+        itemProperties.type = 'boolean';
+      }
+
+      if (field.type === FieldMetadataType.RELATION) {
+        itemProperties.type = 'array';
+        itemProperties['items'] = {
+          type: 'object',
+          properties: {
+            node: {
+              type: 'object',
+            },
+          },
+        };
+      }
+
+      if (Object.keys(field.targetColumnMap).length > 1) {
+        itemProperties.type = 'object';
+        itemProperties['properties'] = Object.keys(
+          field.targetColumnMap,
+        ).reduce((properties, key) => {
+          properties[key] = { type: 'string' };
+
+          return properties;
+        }, {});
```

Comment:
@martmull Properly type the reduce function 
```suggestion
        }, {} as Record<string, { type: 'string'; }>);
```